Automattic/jetpack
# Social: Remove the editor tweetstorm components

## Proposed changes:

@ellatrix let us know that the TweetDivider component was causing performance issues in the browser, and addressed them in #33721.

In discussion about it, we decided to remove the components entirely as the feature can't be used while we don't have access to the twitter API.

This PR removes the components, ~we should follow up with removing the tweetstorm functionality from the `publicize` data store.~ Turns out that a lot of stuff was coupled together, so this also removes code from the store, as well as the the `gather-tweetstorm` editor extension, which could be another performance problem.

### Other information:


## Jetpack product discussion

p1697840484666559-slack-C02JJ910CNL


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:
* Ensure auto-sharing is enabled on a Jetpack site or use a WordPress.com site
* Load the editor
* Publish a test post
* Check the post is shared correctly
* Check for no related console errors in the browser
